tools/incubator.js

```diff
@@ -209,6 +209,22 @@ const Incubator = new Lang.Class({
             basin_manifest['content'].push(this._import_video(asset));
         });
 
+
+        /* Once we're done, look over all the content in the manifest and
+         * set the parent key on each metadata struct. */
+        let manifest_table = {};
```

Comment:
It would be better to use a `Map` or possibly `WeakMap` here.

Reply:
Yep - also it allows the usage of `reduce` as well quite nicely :)